### PR TITLE
controllers/migration: support inlined migration dir

### PR DESCRIPTION
**Pull request by datdao**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -177,6 +177,34 @@ Now, let's try versioned migrations with a PostgreSQL database.
       configMapRef:
         name: "migrationdir"
   ```
+
+  We can define a migration directory inlined in the migration resource instead of using a ConfigMap:
+  
+  ```hcl
```
Comment (rotemtam) [inline README.md:183]:
```suggestion
  ```yaml
```

File: controllers/atlasmigration_controller_test.go [hunk 2]
```diff
@@ -295,12 +295,32 @@ func TestReconcile_extractCloudMigrationData(t *testing.T) {
 	cleanUp()
 }
 
-func TestReconcile_createTmpDir(t *testing.T) {
+func TestReconcile_createTmpDirFromMap(t *testing.T) {
+	tt := newMigrationTest(t)
+
+	// When the configmap exists
+	dir, cleanUp, err := tt.r.createTmpDirFromMap(context.Background(), map[string]string{
```
Comment (rotemtam) [inline controllers/atlasmigration_controller_test.go:302]: I prefer not to write unit tests from private functions. They aren't part of the API and testing them directly reduces flexibility. Find a way to test the behavior using the components exposed API.

**Review by rotemtam — COMMENTED** [review]
Looks good, see comments.

File: controllers/atlasmigration_controller_test.go (hunk 2 above)
Comment (datdao) [reply controllers/atlasmigration_controller_test.go:338]: I have added more tests

**Review by rotemtam — APPROVED** [review]
Please create a release after merging so the docs are synced with the latest release. Well done!

**Pull request merged**